### PR TITLE
document docker-in-docker

### PR DESCRIPTION
**Pull request by technosophos**
Document using Docker-in-Docker

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: docs/content/topics/scripting_advanced.md [hunk 1]
```diff
@@ -193,3 +193,34 @@ world
...
+
+
+## Using Docker Within a Brigade Job
+
+It is possible to use Docker inside of a Brigade job. However, you will need to do some extra work. Because a Job must run in privileged mode to use the Docker socket, the method here presents a security risk and should not be allowed for untrusted brigade scripts.
+
+Before you can write scripts that use privileged mode, you will need to set the following permissions on your Brigade project:
+
+```yaml
+allowHostMounts: "true"
```
Comment (vdice) [inline docs/content/topics/scripting_advanced.md]: Strictly speaking, `allowHostMounts: "true"`/`dind.docker.enabled =true;` wouldn't necessarily be required for the example below, as the job container runs the docker daemon itself and only really requires `privileged` be set.  A bit of clarification will be added to the review here...

**Comment by krancour** [comment]
Echoing and amending some of @vdice's comments above...

Mounting the host's docker socket is commonly called "dood" (Docker-outside-of-Docker) and doesn't require a privileged container.

"dind" (Docker-in-Docker) requires a privileged container and doesn't required mounting the host's docker socket.

The latter is generally considered the cleaner and safer approach, although use cases for the former exist. It's _might_ be worth documenting and providing examples for both.

File: docs/content/topics/scripting_advanced.md (hunk 1 above)
Comment (technosophos) [reply docs/content/topics/scripting_advanced.md]: Ooo... yeah, it appears that docker.enabled = true actually breaks it

File: docs/content/topics/scripting_advanced.md [hunk 2]
```diff
@@ -193,3 +193,32 @@ world
...
+
+To use Docker-in-Docker inside of a job, you will need to do three things:
+
+- Select a container image for your job that can use Docker in Docker
+- Set the job to `privileged = true`
+- Run extra tasks to setup Docker-in-Docker (see the `dockerd-entrypoint.sh &` command below)
+
+```javascript
+  let dind = new Job("dind-run", "docker:dind");
+  dind.privileged = true; // allowPriviledgeJobs must be set to true for this to work
```
Comment (vdice) [inline docs/content/topics/scripting_advanced.md]: typo in `allowPriviledgeJobs`

Comment (technosophos) [reply docs/content/topics/scripting_advanced.md]:
```suggestion
  dind.privileged = true; // allowPrivilegedJobs must be set to true for this to work
```

**Pull request merged**